### PR TITLE
Rename Grammar -> Rules and use Grammar node instead of a dictionary

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

File: pegen/testutil.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def parse_string(source, parser_class, *, dedent=True, verbose=False):
 
 def make_parser(source):
     # Combine parse_string() and generate_parser().
-    rules = parse_string(source, GrammarParser).rules
+    rules = parse_string(source, GrammarParser)
```
Comment (gvanrossum) [inline pegen/testutil.py]: I think this returns a `Grammar` so the variable should be named `grammar`.

File: test/test_pegen.py [hunk 2]
```diff
@@ -531,7 +535,7 @@ def test_parse_or_grammar(self):
 
         visitor.visit(rules)
 
-        # Rules/Rule/Rhs/Alt/NamedItem/NameLeaf   -> 6
+        # Grammar/Rule/Rhs/Alt/NamedItem/NameLeaf   -> 6
```
Comment (gvanrossum) [inline test/test_pegen.py]: Fix ASCII art alignment?

File: pegen/build.py [hunk 3]
```diff
@@ -79,9 +79,9 @@ def build_generator(
     with open(output_file, "w") as file:
         gen: ParserGenerator
         if output_file.endswith(".c"):
-            gen = CParserGenerator(rules.rules, file)
+            gen = CParserGenerator(rules, file)
```
Comment (gvanrossum) [inline pegen/build.py]: Looks like this argument should be called `grammar`. (Also in the `def` line above and for `PythonParserGenerator` below).

**Comment by pablogsal** [comment]
Thanks for the fixes!

**Pull request merged**